api/api.go

```diff
@@ -108,6 +111,69 @@ func NewApi(clientset kubernetes.Interface, fasitUrl, clusterDomain, clusterName
 	}
 }
 
+func (api Api) deploy(w http.ResponseWriter, r *http.Request) *appError {
+	requests.With(prometheus.Labels{"path": "deploy"}).Inc()
+
+	deploymentRequest, err := unmarshalDeploymentRequest(r.Body)
+
+	if err != nil {
+		return &appError{err, "unable to unmarshal deployment request", http.StatusBadRequest}
+	}
+
+	//TODO remove this once grace period ends
+	deploymentRequest, warnings := ensurePropertyCompatability(deploymentRequest)
+
+	fasit := FasitClient{api.FasitUrl, deploymentRequest.Username, deploymentRequest.Password}
+
+	glog.Infof("Starting deployment. Deploying %s:%s to %s\n", deploymentRequest.Application, deploymentRequest.Version, deploymentRequest.Environment)
+
+	manifest, err := GenerateManifest(deploymentRequest)
+	if err != nil {
+		return &appError{err, "unable to generate manifest/nais.yaml", http.StatusInternalServerError}
+	}
+
+	var fasitEnvironmentClass string
+
+	if hasResources(manifest) {
+		if deploymentRequest.FasitEnvironment == "" {
+			return &appError{err, "no fasit environment provided, but contains resources to be consumed or exposed", http.StatusInternalServerError}
+		}
+		if err := validateFasitRequirements(fasit, deploymentRequest.Application, deploymentRequest.FasitEnvironment); err != nil {
+			return &appError{err, "validating requirements for deployment failed", http.StatusInternalServerError}
+		}
+		fasitEnvironmentClass, err = fasit.GetFasitEnvironmentClass(deploymentRequest.FasitEnvironment)
+	}
+
+	glog.Infof("Starting deployment. Deploying %s:%s to %s\n", deploymentRequest.Application, deploymentRequest.Version, deploymentRequest.Environment)
+
+	naisResources, err := FetchFasitResources(fasit, deploymentRequest.Application, deploymentRequest.FasitEnvironment, deploymentRequest.Zone, manifest.FasitResources.Used)
+	if err != nil {
+		return &appError{err, "unable to fetch fasit resources", http.StatusBadRequest}
+	}
+
+	if api.IstioEnabled && !manifest.Istio.Disabled {
```

Comment:
Why not `manifest.Istio.Enabled`? Easier to read.

Reply:
because the zero-value of bool is false, so we want to default to this. Also our merge library doesnt handle the opposite case when we set the default values

Reply:
Ah, had forgot that.